auction9-frontend/src/app/add-update-auction-dialog/add-update-auction-dialog.component.ts

```diff
@@ -0,0 +1,166 @@
+        Validators.required,
+        Validators.maxLength(300)
+      ]),
+      price: new FormControl(this.update ? auction.price : '', [
+        Validators.required,
+        Validators.pattern("^[1-9]\\d*(,\\d+)?$"), //only numbers and comma regex
+        Validators.maxLength(10),
+      ]),
+      dateGroup: new FormGroup({
+        startDate: new FormControl(this.update ? moment(auction.date_from).format() : '', [
+          Validators.required
+        ]),
+        endDate: new FormControl(this.update ? moment(auction.date_to).format() : '', [
+          Validators.required
+        ]),
+      }),
+      startTime: new FormControl(this.update ? moment(auction.date_from).format("HH:mm") : '', [
+        Validators.required
+      ]),
+      endTime: new FormControl(this.update ? moment(auction.date_to).format("HH:mm") : '', [
+        Validators.required
+      ]),
+    }, {
+      validators: [
+        timeValidator
+      ],
+      updateOn: "change"
+    });
+  }
+
+  ngOnInit(): void {
+  };
+
+  setDates() {
+    let startDate = this.addOrUpdateAuctionForm.get('dateGroup.startDate').value;
+    let endDate = this.addOrUpdateAuctionForm.get('dateGroup.endDate').value;
+    let startTime = this.addOrUpdateAuctionForm.controls.startTime.value;
+    let endTime = this.addOrUpdateAuctionForm.controls.endTime.value;
+
+    // Formatting start and end date
+    let formattedStartDate = moment(startDate).format("YYYY-MM-DD");
+    let formattedEndDate = moment(endDate).format("YYYY-MM-DD");
+
+    // Concatenating date and time
+    let concStartDateTime = moment(formattedStartDate + " " + startTime);
+    let concEndDateTime = moment(formattedEndDate + " " + endTime);
+
+    // Formatting concatenated date and time
+    let startDateTime = moment(concStartDateTime).format("YYYY-MM-DD HH:mm:ss");
+    let endDateTime = moment(concEndDateTime).format("YYYY-MM-DD HH:mm:ss");
+
+    return { startDateTime, endDateTime };
+  }
+
+
+  onCreateAuctionClick() {
+    if (this.addOrUpdateAuctionForm.invalid) {
+      return;
+    }
+    else {
+      let startDateTime = this.setDates().startDateTime;
+      let endDateTime = this.setDates().endDateTime;
+
+      // Until we implement user sign in, created_by will be hardcoded user with id=1
+      this.auctionService.addAuction(
+        {
+          "title": this.addOrUpdateAuctionForm.controls.title.value,
+          "description": this.addOrUpdateAuctionForm.controls.description.value,
+          "price": this.addOrUpdateAuctionForm.controls.price.value,
+          "date_from": startDateTime,
+          "date_to": endDateTime,
+          "created_by": 1
+        }).subscribe(() => {
+          this.snackBar.open('Auction created successfully.', '',
+            {
+              duration: 2000,
+              panelClass: ['light-snackbar']
+            });
+        }), (err: any) => {
+          this.snackBar.open('Unable to create auction.', '',
```

Comment:
We should surface the reason, if any. Not in this PR

Reply:
Okay.